### PR TITLE
Eliminate trailing spaces w/o descriptions

### LINKED ISSUE #219: Faulty generation of markdown format for data type list
## Describe the Bug
We are writing heavily custom data types to validate special hash formats. We are using `puppet strings generate --format markdown --out REFERENCES.md` for documentation. Our markdown-lint CI test complains about the `MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]`.

## Expected Behavior
Generating a markdown formated REFERENCES.md withtout `MD009/no-trailing-spaces` issues

## Steps to Reproduce
- write a module with a data type in it 
- add the documentation header to the data type file without `@summary ... `
- install markdown-lint:
  - npm install -g markdownlint-cli
  - npm install -g markdownlint
- execute the markdownlint:
  - node_modules/.bin/markdownlint <path_to_markdown_file>

## Environment
 - Version: 2.3.1
 - Platform: any (CI: ubuntu-18.04, local: MacOS 10.15.2)

## Additional Context
%

### PR DESCRIPTION
**Pull request by binford2k**
Fixes #219

### CODE REVIEW

File: lib/puppet-strings/markdown/templates/table_of_contents.erb [hunk 1]
```diff
@@ -6,20 +6,20 @@ _Public <%= group_name %>_
 
 <% group.each do |item| -%>
 <% unless item[:private] -%>
-* [`<%= item[:name] %>`](#<%= item[:link] %>): <%= item[:desc] %>
+* [`<%= item[:name] %>`](#<%= item[:link] %>)<% unless item[:desc].empty? %>: <%= item[:desc] %><% end %>
```
Comment (binford2k) [inline lib/puppet-strings/markdown/templates/table_of_contents.erb:9]: Is `item[:desc]` guaranteed to not be `nil`?

Comment (scotje) [reply lib/puppet-strings/markdown/templates/table_of_contents.erb:9]: > Is `item[:desc]` guaranteed to not be `nil`?

I wouldn't count on it...

```suggestion
* [`<%= item[:name] %>`](#<%= item[:link] %>)<% unless item[:desc].nil? || item[:desc].empty? %>: <%= item[:desc] %><% end %>
```

File: lib/puppet-strings/markdown/templates/table_of_contents.erb [hunk 2]
```diff
@@ -6,20 +6,20 @@ _Public <%= group_name %>_
...
+* [`<%= item[:name] %>`](#<%= item[:link] %>)<% unless item[:desc].nil? || item[:desc].empty? %>: <%= item[:desc] %><% end %>
 <% end -%>
 <% end -%>
 
 _Private <%= group_name %>_
 
 <% group.each do |item| -%>
 <% if item[:private] -%>
-* `<%= item[:name] %>`: <%= item[:desc] %>
+* `<%= item[:name] %>`<% unless item[:desc].empty? %>: <%= item[:desc] %><% end %>
```
Comment (binford2k) [inline lib/puppet-strings/markdown/templates/table_of_contents.erb:17]:
```suggestion
* `<%= item[:name] %>`<% unless item[:desc].nil? || item[:desc].empty? %>: <%= item[:desc] %><% end %>
```

File: lib/puppet-strings/markdown/templates/table_of_contents.erb [hunk 3]
```diff
@@ -6,20 +6,20 @@ _Public <%= group_name %>_
...
 <% end -%>
 <% end -%>
 <% else -%>
 <% group.each do |item| -%>
-* [`<%= item[:name] %>`](#<%= item[:link] %>): <%= item[:desc] %>
+* [`<%= item[:name] %>`](#<%= item[:link] %>)<% unless item[:desc].empty? %>: <%= item[:desc] %><% end %>
```
Comment (binford2k) [inline lib/puppet-strings/markdown/templates/table_of_contents.erb:22]:
```suggestion
* [`<%= item[:name] %>`](#<%= item[:link] %>)<% unless item[:desc].nil? || item[:desc].empty? %>: <%= item[:desc] %><% end %>
```

**Pull request merged**